File: RA1/JAVA/EjemploHerencia/Animal.java
package EjemploHerencia;

public class Animal {
    //Atributos
    protected String name;

    //Métodos
    public Animal(String name){
        super();
        this.name = name;
    }

    public void eat(String something){
        System.out.println(this.name + " come " + something);
    }

    public void makeSound(){
        System.out.println(this.name + " hace un sonido...");
    }

    //getter y setter
    //getter permite obtener información de un atributo, se representa con get<nombreAtributo>
    public String getName(){
        return this.name;
    }

    //setter permite modificar un atributo, se representa con set<nombreAtributo>
    public void setName(String newName){
        if(newName.equals(null) || newName.equals(""))
            System.out.println("Set name is denial...");
        else
            this.name = newName;
    }
}

File: RA1/PYTHON/ejemplo_herencia/animal.py
class Animal:
    #constructor
    def __init__(self, name):
        self.name = name

    def eat(self, something):
        print(self.name + " come " + something)

    def make_sound(self):
        print(self.name + " hace un sonido... ")

    def get_name(self):
        return self.name

    def set_name(self, new_name):
        if new_name == None or new_name == "":
            print("Set name es denial...")
        else:
            self.name = new_name